lms/djangoapps/ccx/views.py

```diff
@@ -223,9 +223,14 @@ def override_fields(parent, data, graded, earliest=None, ccx_ids_to_delete=None)
                 ccx_ids_to_delete.append(get_override_for_ccx(ccx, block, 'start_id'))
                 clear_ccx_field_info_from_ccx_map(ccx, block, 'start')
 
-            due = parse_date(unit['due'])
-            if due:
-                override_field_for_ccx(ccx, block, 'due', due)
+            # Only subsection (aka sequential) has due date
+            if 'due' in unit:
```

Comment:
@sarina here i am checking that key ```due``` exist in dict ```unit```.
Because only units and subsections has due date.
In sum due date will be missing in case if section aka chapter. I also did update comments.